ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -794,6 +827,128 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
+	tcAddQdiscRootCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddQdiscRootCommandString, interfaceName)
+	cmdList := strings.Split(tcAddQdiscRootCommandComposed, " ")
+	cmdOutput, err := h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		logger.Info(fmt.Sprintf("'%s' command failed with the following error: '%s'. std output: '%s'",
+			tcAddQdiscRootCommandComposed, err, string(cmdOutput[:])))
+		return err
+	}
+	logger.Info(fmt.Sprintf("'%s' command result: '%s'", tcAddQdiscRootCommandComposed, string(cmdOutput[:])))
+	tcAddQdiscLossCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddQdiscLossCommandString, interfaceName, lossPercent)
+	cmdList = strings.Split(tcAddQdiscLossCommandComposed, " ")
+	cmdOutput, err = h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		logger.Info(fmt.Sprintf("'%s' command failed with the following error: '%s'. std output: '%s'",
+			tcAddQdiscLossCommandComposed, err, string(cmdOutput[:])))
+		return err
+	}
+	logger.Info(fmt.Sprintf("'%s' command result: '%s'", tcAddQdiscLossCommandComposed, string(cmdOutput[:])))
+	// After creating the queueing discipline, create filters to associate the IPs in the request with the handle.
+	for _, ip := range request.Sources {
+		tcAddFilterForIPCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddFilterForIPCommandString, interfaceName, *ip)
+		cmdList = strings.Split(tcAddFilterForIPCommandComposed, " ")
+		_, err = h.runExecCommand(ctx, cmdList)
+		if err != nil {
+			logger.Info(fmt.Sprintf("'%s' command failed with the following error: '%s'. std output: '%s'",
+				tcAddFilterForIPCommandComposed, err, string(cmdOutput[:])))
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkTCFault check if there's existing network-latency fault or network-packet-loss fault.
+func (h *FaultHandler) checkTCFault(ctx context.Context, taskMetadata *state.TaskResponse) (bool, bool, error) {
+	interfaceName := taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].NetworkInterfaces[0].DeviceName
+	networkMode := taskMetadata.TaskNetworkConfig.NetworkMode
+	// If task's network mode is awsvpc, we need to run nsenter to access the task's network namespace.
+	nsenterPrefix := ""
+	if networkMode == ecs.NetworkModeAwsvpc {
+		nsenterPrefix = fmt.Sprintf(nsenterCommandString, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+	}
+
+	// We will run the following Linux command to assess if there existing fault.
+	// "tc -j q show dev {INTERFACE} parent 1:1"
+	// The command above gives the output of "tc q show dev {INTERFACE} parent 1:1" in json format.
+	// We will then unmarshall the json string and evaluate the fields of it.
+	tcCheckInjectionCommandComposed := nsenterPrefix + fmt.Sprintf(tcCheckInjectionCommandString, interfaceName)
+	cmdList := strings.Split(tcCheckInjectionCommandComposed, " ")
+	cmdOutput, err := h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		logger.Info(fmt.Sprintf("'%s' command failed with the following error: '%s'. std output: '%s'",
```

Comment:
Same here

Reply:
Published a new revision to address a test failure, also addressed this.